app/assets/javascripts/hyrax/turbolinks_events.js

```diff
@@ -6,3 +6,20 @@ $(document).on('turbolinks:load', function() {
   // See https://github.com/samvera/hyrax/issues/2906
   window.embedScriptIncluded = false;
 });
+
+// Fixes a back/forward navigation problem with UV and turbolinks
+// See https://github.com/samvera/hyrax/issues/2964
+// This is based on https://github.com/turbolinks/turbolinks/issues/219#issuecomment-275838923
+$(window).on('popstate', function(event) {
+  var turbolinks_location = Turbolinks.Location.wrap(window.location);
+  if (Turbolinks.controller.location.requestURL === turbolinks_location.requestURL) {
```

Comment:
Does `turbolinks_location` always return a value?   If it's `undefined`, it'll throw an error.  If it might not always be an object, you could test that it is an object and/or has the `requestURL` property.  

Other than that, the JS looks clean to me.

Reply:
I think it will always be an object since `window.location` should always be an object and `wrap` creates a new object around it.